src/components/SearchBar/SearchBar.hook.ts

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect, type ChangeEvent, type KeyboardEvent } from 'react';
+
+import { DEBOUNCE_DELAY } from './SearchBar.constants';
+
+import type { SearchBarProps } from './SearchBar.types';
+
+export const useSearchBar = (props: SearchBarProps) => {
+  const {
+    value: controlledValue,
+    defaultValue,
+    onSearch,
+    debounceDelay = DEBOUNCE_DELAY,
+    onChange,
+  } = props;
+
+  const isControlled = controlledValue !== undefined;
+  const [internalValue, setInternalValue] = useState(defaultValue || '');
```

Comment:
setInternalValue is initialized with a string (defaultValue || ''), so internalValue's inferred type is string. Later you call setInternalValue(controlledValue) where controlledValue can be undefined (props.value is optional). That will cause a TypeScript error or unsafe state assignment. Ensure you never pass undefined to the state setter or type the state to accept undefined explicitly.

**Sugerencia:**
```typescript
Type internal state explicitly and normalize controlledValue when syncing. For example:
const [internalValue, setInternalValue] = useState<string>(defaultValue ?? '');
...
if (isControlled) {
  setInternalValue(controlledValue ?? '');
}
```